src/olympia/git/management/commands/migrate_git_storage_to_new_structure.py

```diff
@@ -0,0 +1,131 @@
+        parser.add_argument(
+            '--dont-migrate',
+            action='store_true',
+            help=('Only create the new directory structure, do not migrate anything.'),
+        )
+        parser.add_argument(
+            '--dont-rename-root',
+            action='store_true',
+            help=('Migrate, but do not rename root directory at the end.'),
+        )
+        parser.add_argument(
+            '--fake',
+            action='store_true',
+            help=('Fake all migration/rename operations.'),
+        )
+
+    def get_new_temporary_file_storage_path(self):
+        """Temporary name for the new storage path.
+
+        Once everything is migrated the old directory will be removed and we'll
+        rename this one, removing the 'new-' prefix."""
+        return os.path.join(settings.STORAGE_ROOT, 'new-git-storage')
+
+    def get_old_temporary_file_storage_path(self):
+        """Temporary name for the old storage path.
+
+        Used at the very end, once we've migrated everything and we're about to
+        rename the storage path."""
+        return os.path.join(settings.STORAGE_ROOT, 'old-git-storage')
+
+    def handle(self, *args, **options):
+        self.verbosity = int(options.get('verbosity', 0))
+        self.fake = bool(options.get('fake'))
+        self.print_prefix = '[fake] ' if self.fake else ''
+        old_git_file_storage_path = self.get_old_temporary_file_storage_path()
+        if os.path.exists(old_git_file_storage_path):
+            raise CommandError(f'{old_git_file_storage_path} should not exist')
+        self.create_new_directory_structure()
+        if options.get('dont_migrate'):
+            if self.verbosity:
+                self.stdout.write('Not migrating per --dont-migrate parameter.\n')
+            return
+        self.migrate()
+        if options.get('dont_rename_root'):
+            if self.verbosity:
+                self.stdout.write(
+                    'Not renaming top directory per --dont-rename-root parameter.\n'
+                )
+            return
+        self.rename_top_directory()
+
+    def get_new_path(self, addon_id):
+        return os.path.join(
+            self.get_new_temporary_file_storage_path(), id_to_path(addon_id, depth=3)
+        )
+
+    def create_new_directory_structure(self):
+        new_file_storage_path = self.get_new_temporary_file_storage_path()
+        for x in range(0, 10):
+            for y in range(0, 10):
+                # We only have a single add-on with an id < 3 digits, and its
+                # id is 60, let's add it manually.
+                zrange = list(range(0, 10))
+                if x == 0 and y == 6:
+                    zrange.append(60)
+                for z in zrange:
+                    path = os.path.join(
+                        new_file_storage_path,
+                        f'{x}',
+                        f'{y}{x}',
+                        f'{z}{y}{x}' if z < 10 else f'{z}',
```

Comment:
this is special casing, _just_ for id=60, right?  why not just create it with `os.makedirs(os.path.join(new_file_storage_path, '0', '60', '60'))` directly first?

Reply:
In hindsight... Yeah that's better :) I didn't know how many special cases like that I'd have to handle when I started, so I figured I would handle them all in a generic way, but since we have only one... I'll change it

Reply:
Special casing is gone with the ne 100x100 approach in https://github.com/mozilla/addons-server/pull/18906/commits/26e45049b229182ed35217fae485da6123137cd2